### PR TITLE
rename cluster-osin-operator/cluster-authentication-operator

### PR DESCRIPTION
**Pull request by sallyom**
@enj @evb changed the import paths here, too
`cluster-osin-operator` to `cluster-authentication-operator`
and using the naming conventions laid out by @deads2k in email thread:
```
ClusterOperator name  "authentication" - see note below
Operator Namespace “openshift-authentication-operator”   - this PR
Operator Deployment “cluster-authentication-operator”  - this PR
Operator Git repo “openshift/cluster-authentication-operator”  - will be changed
Operand Namespace “openshift-authentication” - this PR
```
*note:  - in this PR it's "cluster-authentication-operator".  There isn't consistency with that 
name in clusters today, although going with majoritiy it would be "openshift-authentication-operator"

### CODE REVIEW

**Comment by ericavonb** [comment]
@sallyom want to rebase? let's get this thing merged!

File: pkg/operator2/operator.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func (c *authOperator) handleSync(configOverrides []byte) error {
 
 func defaultLabels() map[string]string {
 	return map[string]string{
-		"app": "origin-cluster-osin-operator2",
+		"app": "origin-cluster-authentication-operator2",
```
Comment (enj) [inline pkg/operator2/operator.go]: we don't need the `2`

File: pkg/operator2/operator.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 )
 
 const (
-	targetName = "openshift-osin" // TODO fix
+	targetName = "openshift-integrated-auth"
```
Comment (enj) [inline pkg/operator2/operator.go]: can you outline the new name requirements so we know if this matches

File: pkg/cmd/operator2/cmd.go [hunk 3]
```diff
@@ -9,8 +9,8 @@ import (
 )
 
 const (
-	componentName      = "cluster-osin-operator2"
-	componentNamespace = "openshift-core-operators"
+	componentName      = "cluster-authentication-operator2"
+	componentNamespace = "openshift-authentication-operator"
 )
 
 func NewOperator() *cobra.Command {
```
Comment (enj) [inline pkg/cmd/operator2/cmd.go]: cmd.User needs fixing

File: pkg/cmd/operator2/cmd.go [hunk 4]
```diff
@@ -9,8 +9,8 @@ import (
 )
 
 const (
-	componentName      = "cluster-osin-operator2"
-	componentNamespace = "openshift-core-operators"
+	componentName      = "cluster-authentication-operator2"
```
Comment (enj) [inline pkg/cmd/operator2/cmd.go]: drop 2

File: pkg/cmd/operator2/cmd.go [hunk 5]
```diff
@@ -9,8 +9,8 @@ import (
...
+	componentNamespace = "openshift-authentication-operator"
```
Comment (enj) [inline pkg/cmd/operator2/cmd.go]: cite requirements

File: pkg/cmd/operator/cmd.go [hunk 6]
```diff
@@ -9,8 +9,8 @@ import (
 )
 
 const (
-	componentName      = "cluster-osin-operator"
-	componentNamespace = "openshift-core-operators"
+	componentName      = "cluster-authentication-operator"
```
Comment (enj) [inline pkg/cmd/operator/cmd.go]: do not touch anything in `pkg/operator`

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml [hunk 7]
```diff
@@ -1,27 +1,27 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  namespace: openshift-core-operators
-  name: origin-cluster-osin-operator
+  namespace: openshift-authentication-operator
```
Comment (enj) [inline manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: do not mess with the original operator, we have no tests

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml [hunk 8]
```diff
@@ -1,27 +1,27 @@
...
+        app: origin-cluster-authentication-operator
     spec:
-      serviceAccountName: origin-cluster-osin-operator
+      serviceAccountName: origin-cluster-authentication-operator
       containers:
       - name: operator
         image: quay.io/openshift/origin-cluster-authentication-operator:latest
         imagePullPolicy: IfNotPresent
-        command: ["osin-operator", "operator"]
+        command: ["authentication-operator", "operator"]
```
Comment (enj) [inline manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: this is the only line I expect to change

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml [hunk 9]
```diff
@@ -41,27 +41,27 @@ spec:
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  namespace: openshift-core-operators
-  name: origin-cluster-osin-operator2
+  namespace: openshift-authentication-operator
```
Comment (enj) [inline manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: lots of 2's that need to die

**Comment by enj** [comment]
I do not think the risk of breaking operator1 is worth it, so I do no think we should touch the old operator at all.  At the bare minimum https://github.com/openshift/origin/pull/21733 needs to be merged before this.

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml (hunk 9 above)
Comment (sallyom) [reply manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: removed them

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml (hunk 8 above)
Comment (sallyom) [reply manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: done

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml (hunk 7 above)
Comment (sallyom) [reply manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: ok, done

File: pkg/cmd/operator/cmd.go (hunk 6 above)
Comment (sallyom) [reply pkg/cmd/operator/cmd.go]: ok, done

File: pkg/cmd/operator2/cmd.go (hunk 4 above)
Comment (sallyom) [reply pkg/cmd/operator2/cmd.go]: done

File: manifests/0000_09_cluster-authentication-operator_05_deploy.yaml (hunk 8 above)
Comment (sallyom) [reply manifests/0000_09_cluster-authentication-operator_05_deploy.yaml]: so for both deployments, should be same, right? 
`command: ["authentication-operator", "operator"]`   (also in pkg/cmd/operator|operator2/cmd.go, same cmd.User for both? `operator`?)

File: pkg/cmd/operator2/cmd.go (hunk 3 above)
Comment (sallyom) [reply pkg/cmd/operator2/cmd.go]: I've changed to `operator`

File: pkg/operator2/operator.go (hunk 1 above)
Comment (sallyom) [reply pkg/operator2/operator.go]: done, fixed

File: pkg/operator2/operator.go (hunk 2 above)
Comment (sallyom) [reply pkg/operator2/operator.go]: from email thread (kube-apiserver, naming conventions from @deads2k  are as follows:
```
ClusterOperator name  "authentication" 
Operator Namespace “openshift-authentication-operator”   -good
Operator Deployment “cluster-authentication-operator”  -good
Operator Git repo “openshift/cluster-authentication-operator”  -will be changed
Operand Namespace “openshift-authentication” -edited  this PR to match
```

File: pkg/cmd/operator2/cmd.go (hunk 5 above)
Comment (sallyom) [reply pkg/cmd/operator2/cmd.go]: I've placed naming requirements from email thread in commit msg for reference.

**Comment by enj** [comment]
#34

**Pull request closed without merging**